Panics during compression

Of my ~20k replays, a dozen failed to compress due to a panic: `slice index starts at <large number, around 450> but ends at 15` which is the slicing done [here](https://github.com/AlexanderHarrison/slpz/blob/afed6b7147e16d8f4fd686aaa7f9a40285e81044/src/lib.rs#L136C39-L136C58). It seems that in those cases `raw_len` is 0 which is why `metadata_offset` ends up being 15 instead of something reasonable.

Most of the replays are tiny stubs that the slippi client also chokes on. Here's [one of them](https://github.com/user-attachments/files/17895342/Game_20201012T045125.slp.zip) just for reference.

The [other](https://github.com/user-attachments/files/17895477/Game_20191014T235103.slp.zip) [two](https://github.com/user-attachments/files/17895478/Game_20191031T225433.slp.zip) are both from October 2019 and both ended with a quit-out or crash. Even though I can play each of them to completion in the slippi client, `slippilab`, `peppi`, `slippc` also fail to parse them:

peppi output:
```
[ERROR] I/O error: failed to fill whole buffer
```

slippc output:
```
  WARNING:     0-byte raw data detected; replay may be corrupt
  WARNING:     Event byte offset exceeds raw data length; replay may be corrupt
  WARNING:     Don't know what's happening; expected key
  WARNING:   Failed to parse metadata
  WARNING:   No game end event found; replay may be corrupt
  WARNING:   Replay parsed with 4 errors
    ERROR:     0-byte raw data detected; cannot continue parsing
    ERROR:   Failed to parse header
    ERROR:   Failed to encode input; exiting
```

It's no big deal that `slpz` fails to read those malformed replays, but if possible it'd be nice for it to skip them (with a more instructive error) rather than crashing, especially for large batch conversion with `-d`. I had to go find which files were causing crashes and move them to get it to progress to other valid files.

Feel free to close this if it's deemed too much of a random edge case. I know most people don't have 5 year old corrupt slp's sitting around and it's literally a one in a thousand failure mode, so idk if it'd actually be useful to worry about.

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -90,6 +90,7 @@ pub fn compress(compressor: &mut Compressor, slp: &[u8]) -> Result<Vec<u8>, Comp
 
     // get metadata
     let raw_len = u32::from_be_bytes(slp[11..15].try_into().unwrap()) as usize;
+    if raw_len == 0 { return Err(CompError::InvalidFile) };
     let metadata_offset = 15+raw_len;
     let metadata = &slp[metadata_offset..];
 

